Should we fix `-Wc++20-extensions`?

```
--- stderr: hardware_interface                                
/workspaces/ros2_rolling_ws/src/ros2_control/hardware_interface/src/resource_manager.cpp:1024:87: warning: captured structured bindings are a C++20 extension [-Wc++20-extensions]
 1024 |           const auto limiter_fn = [this, joint_name, interface_name, desired_period, &limiters](
      |                                                                                       ^
/workspaces/ros2_rolling_ws/src/ros2_control/hardware_interface/src/resource_manager.cpp:1003:29: note: 'limiters' declared here
 1003 |       for (auto & [hw_name, limiters] : joint_limiters_interface_)
      |                             ^
1 warning generated.
```

--- a/hardware_interface/src/resource_manager.cpp
+++ b/hardware_interface/src/resource_manager.cpp
@@ -1003,8 +1003,10 @@ public:
   {
     if (interface)
     {
-      for (auto & [hw_name, limiters] : joint_limiters_interface_)
+      for (auto & entry : joint_limiters_interface_)
       {
+        auto & limiters = entry.second;
+
         // If the prefix is a joint name, then bind the limiter to the command interface
         if (limiters.find(interface->get_prefix_name()) != limiters.end())
         {

